### PR TITLE
refine: add obj_range'_disjoint

### PR DESCRIPTION
**Pull request by michaelmcinerney**
This also moves several lemmas required for obj_range'_disjoint to Invariants_H

### CODE REVIEW

**Review by lsf37 — APPROVED** [review]
Nice. Looking forward to arch splitting Refine ;-)

**Comment by michaelmcinerney** [comment]
There are some errors in the ARM, ARM_HYP, and X64 tests. I was a bit lazy and was hoping they'd be the same as RISCV, but the object ranges aren't defined using `mask` there. I'll cancel the tests, and push a fix soon

**Comment by michaelmcinerney** [comment]
Apologies for the force push. RISCV has had only a minor style change. ARM, ARM_HYP, and X64 all use the same proofs, which are a little different from the proofs for RISCV. 

The proofs of more than a couple lines that end with `by` unfortunately do take a few seconds to finish.

**Comment by lsf37** [comment]
It looks like there's still a slight misalignment somewhere, we're now getting time outs in CRefine. I remember hitting those kinds of annoyances when we introduced ranges with masks. Should not be too bad, though, hopefully.

File: proof/refine/ARM/Invariants_H.thy [hunk 1]
```diff
@@ -3393,4 +3393,50 @@ add_upd_simps "invs' (gsUntypedZeroRanges_update f s)"
   (obj_at'_real_def)
 declare upd_simps[simp]
 
+lemma neq_out_intv:
+  "\<lbrakk> a \<noteq> b; b \<notin> {a..a + c - 1} - {a} \<rbrakk> \<Longrightarrow> b \<notin> {a..a + c - 1}"
```
Comment (corlewis) [inline proof/refine/ARM/Invariants_H.thy:3397]: I guess this lemma is pretty specific, but do we want to move it to `lib`? That would at least reduce duplication while we haven't arch-split `Refine`.

**Review by corlewis — APPROVED** [review]

**Pull request merged**